File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2022w/ID07IO1/Bridge/src/Shape.java

import java.awt.Graphics;
import java.security.InvalidParameterException;

public abstract class Shape
{
   public abstract void draw(Graphics g);
   public static float[] parseParams(String text,int count)
   {
      String[] params=text.split("[\\s\t,;]+");
      if(count!=params.length) throw new InvalidParameterException(String.format("Bad parameters count expected %d",count));
      float[] result=new float[count];
      for(int i=0;i<count;++i) result[i]=Float.parseFloat(params[i]);
      return result;
   }
}

File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06IO2/ManualDrawing/ManualDrawing/Shape.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;

namespace ManualDrawing
{
    abstract class Shape
    {
        public delegate void SF(float value);
        public static void ParseParams(string text, params Action<float>[] tb)
        {
            int size = tb.Length;
            string[] cells = text.Split("\t ;".ToArray(), StringSplitOptions.RemoveEmptyEntries);
            if (size != cells.Length) throw new Exception("Invalid parameters count");
            for (int i = 0; i < size; ++i) tb[i](float.Parse(cells[i]));
        }
        public static float Scale(float value,float shift,float scale)
        {
            return value * scale + shift;
        }

        public abstract void Draw(Graphics g, PointF shift, float scale);
        public abstract float minX();
        public abstract float maxX();
        public abstract float minY();
        public abstract float maxY();
    }
}
